util/test/sub_test.py

```diff
@@ -261,20 +261,33 @@ def PerfDirFile(s):
 def PerfTFile(test_filename, sfx):
     return test_filename + '.' + perflabel + sfx
 
+def get_chplenv():
+    env_cmd = [os.path.join(utildir, 'printchplenv'), '--all', '--simple', '--no-tidy', '--internal']
+    chpl_env = py3_compat.Popen(env_cmd, stdout=subprocess.PIPE).communicate()[0]
+    chpl_env = dict(map(lambda l: l.split('=', 1), chpl_env.splitlines()))
+    return chpl_env
+
+# os.path.expandvars but stuff chplenv into it too
+def expandvars_chpl(path):
+    orig_env = os.environ.copy()
+    os.environ.update(get_chplenv())
+    new_path = os.path.expandvars(path)
+    os.environ.clear()
+    os.environ.update(orig_env)
+    return new_path
+
```

Comment:
Feels a bit dirty to modify os.environ just to use expandvars. Alternative implementation might be:

```py
# Similar to os.path.expandvars but stuff chplenv into it too
def expandvars_chpl(path):
    expand_env = os.environ.copy()
    expand_env.update(get_chplenv())
    return string.Template(path).substitute(expand_env)
```

but I don't have a strong preference here.

Reply:
Yeah, I have a slight preference for the alternative implementation

Reply:
Changed